blockchain/blockdao.go

```diff
@@ -139,6 +155,16 @@ func (dao *blockDAO) getBlockHeight(hash hash.Hash256) (uint64, error) {
 
 // getBlock returns a block
 func (dao *blockDAO) getBlock(hash hash.Hash256) (*block.Block, error) {
+	if dao.cache != nil {
+		dao.cacheMutex.RLock()
```

Comment:
we may create three private functions to get, insert, and delete cacheBlock, with "defer dao.cacheMutex.RUnlock/Unlock()"